services/platform/platform.go

```diff
@@ -0,0 +1,200 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package platform provides authentication/authorization functionality.
+package platform
+
+import (
+	"context"
+	"net"
+	"os"
+	"time"
+
+	api "github.com/percona-platform/saas/gen/auth/external"
+	"github.com/percona/pmm/utils/tlsconfig"
+	"github.com/percona/pmm/version"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+const (
+	defaultHost            = "check.percona.com:443"
+	defaultRefreshInterval = 24 * time.Hour
+	timeout                = 5 * time.Second
+
+	envHost            = "PERCONA_TEST_AUTH_HOST"
+	envRefreshInterval = "PERCONA_TEST_REFRESH_INTERVAL"
+
+	authType = "PP-v1beta1" // TODO Change to PP-1 after auth API release
+
+)
+
+var ErrNoActiveSessions = errors.New("no active sessions") //nolint:golint
```

Comment:
golint hints that this variable should not be exported

Reply:
Golint asks to add comments to exported variables or make them private. IMHO it's nice to have exported errors, however, comment will repeat the variable name, it's descriptive enough I think.

Reply:
> IMHO it's nice to have exported errors

Why? We are not using it in other packages

Reply:
Maybe I think too generic, but someday someone will see that error as package contract... nevermind. Will make it private)

Reply:
Done